src/Tabs.jsx

```diff
@@ -63,9 +63,15 @@ export class Tabs extends React.Component {
 	}
 }
 Tabs.propTypes = {
-	children: React.PropTypes.arrayOf(
-		React.PropTypes.instanceOf(TabsTab)
-	).isRequired,
+	children: (props, propName, componentName) => {
+		let error;
+		React.Children.forEach(props[propName], child => {
+			if (typeof child === 'undefined' || child.type !== TabsTab) {
+				error = new Error('Children must be React elements of type TabsTab');
```

Comment:
Ah, I see why you kept the `let error` setup since it is set inside a callback.

Just for fun, you can still avoid the `let error` (and mutable state) with:

```js
const validChildren = React.Children.map(
  children,
  child => child instanceof TabsTab
).every(x => x);  // check that all children are TabsTab

if (!validChildren) {
  return new Error(...);
}
```